README.md

```diff
@@ -636,6 +653,136 @@ When defining an Audience, users and groups can be excluded from the audience. T
 
 In the above example, the feature will be enabled for users named `Jeff` and `Alicia`. It will also be enabled for users in the group named `Ring0`. However, if the user is named `Mark`, the feature will be disabled, regardless if they are in the group `Ring0` or not. Exclusions take priority over the rest of the targeting filter.
 
+## Variants
+
+Variants represent a configuration of a feature. A feature flag with variants contains a list of variants and assignment parameters dictating under what conditions each variant should be used. A variant's configuration can be a complex object, but it can also be as simple as a string or number.
+
+``` C#
+public class Variant
+{
+    /// <summary>
+    /// The name of the variant.
+    /// </summary>
+    public string Name { get; set; }
+
+    /// <summary>
+    /// The configuration of the variant.
+    /// </summary>
+    public IConfigurationSection Configuration { get; set; }
+}
+```
+
+### Getting a Feature's Variant
+
+Variants expand on the basic flow of feature management and enable getting a feature's variant to later perform actions based on the name or configuration. This is done through the `IVariantFeatureManager`'s `GetVariantAsync` method.
+
+``` C#
+…
+IVariantFeatureManager featureManager;
+…
+Variant variant = await featureManager.GetVariantAsync(nameof(MyFeatureFlags.FeatureU));
+
+IConfigurationSection variantConfiguration = variant.Configuration;
+
+// Do something with the resulting variant and its configuration
+```
+
+### Setting a Variant's Configuration
+
+For each of the variants in the `Variants` property of a feature, there is a specified configuration. This can be set using either the `ConfigurationReference` or `ConfigurationValue` properties. `ConfigurationReference` is a string path that references a section of the current configuration that contains the feature flag declaration. `ConfigurationValue` is an inline configuration. If both are specified, `ConfigurationValue` is used.
+
+```
+"Variants": [
+    { 
+        "Name": "Big", 
+        "ConfigurationReference": "ShoppingCart:Big" 
+    },  
+    { 
+        "Name": "Small", 
+        "ConfigurationValue": {
+            "Size": 300
+        }
+    } 
+]
+```
+
+### Assigning a Variant
+
+The process of deciding which variant to return for a feature is called assignment. Assignment is determined by the `Allocation` property of the feature. Allocation logic is similar to the [Microsoft.Targeting](./README.md#MicrosoftTargeting) feature filter, with some additional parameters to allow for more specific variant assignment.

```

Comment:
I would avoid mentioning the targeting filter here. It may confuse customers.

It may be helpful, at the end of this section, to explain why the allocation is different from the targeting filter.